app/src/androidTest/java/tech/relaycorp/gateway/background/endpoint/EndpointPreRegistrationServiceTest.kt

```diff
@@ -43,6 +43,7 @@ class EndpointPreRegistrationServiceTest {
     fun setUp() {
         AppTestProvider.component.inject(this)
         runBlocking {
+            localConfig.bootstrap()
```

Comment:
@sdsantos, one of the tests here (`requestPreRegistration`) is failing intermittently because the gateway key pair changes midway through the test because `App.bootstrapGateway()` gets called, so we end up calling `localConfig.bootstrap()` twice. I've just added some logs to verify this.

I'm only adding this call to `localConfig.bootstrap()` because, if I don't, I'd get an exception like this:

```
2021-01-20 17:07:01.992 16499-16535/tech.relaycorp.gateway E/AndroidRuntime: FATAL EXCEPTION: DefaultDispatcher-worker-2
    Process: tech.relaycorp.gateway, PID: 16499
    java.lang.RuntimeException: No key pair was found
        at tech.relaycorp.gateway.domain.LocalConfig.getKeyPair(LocalConfig.kt:31)
        at tech.relaycorp.gateway.domain.endpoint.EndpointRegistration.authorize(EndpointRegistration.kt:29)
        at tech.relaycorp.gateway.background.endpoint.EndpointPreRegistrationService.reply(EndpointPreRegistrationService.kt:73)
    ...
```

How can I make sure that `App.bootstrapGateway()` gets called before the tests run?

Reply:
Usually a way for injecting the scope is ideal, it should already be possible in all classes using Dispatchers.IO, but I didn't design for it from the start.

If do for `waitForCheck` approach, I would say you can remove `localConfig.bootstrap()`, cause it will run anyway in the App class, you just need to wait for it before binding any service.